File: src/webapp/src/models/column.ts
import { merge } from 'lodash';
import { GenericError } from './exceptions';

export class Column {
  id = 0;
  name = '';
  cardState = 0;
  order = 0;

  static create(data: any): Column {
    if (!data) throw new GenericError('data expected');

    const model = new Column();

    merge(model, data);

    return model;
  }
}

File: src/backend/Workboard.Domain/Model/Column.cs
namespace Workboard.Domain.Model;

public partial class Column
{
#nullable disable 
    private Column() { }
#nullable enable

    internal Column(string name, int cardState, int order)
    {
        Name = name;
        CardState = cardState;
        Order = order;
    }

    public int Id { get; private set; }
    public string Name { get; private set; }
    public int CardState { get; private set; }
    public int Order { get; private set; }
}
